lib/spack/spack/test/stage.py

```diff
@@ -728,21 +735,89 @@ def test_first_accessible_path(self, tmpdir):
         # Cleanup
         shutil.rmtree(str(name))
 
+    def test_create_stage_root(self, tmpdir, no_path_access):
+        """Test _create_stage_root permissions."""
+        spack_dir = tmpdir.join('path')
+
+        # Ensure an OS Error is raised on the created, non-user directory
+        with pytest.raises(OSError) as exc_info:
+            spack.stage._create_stage_root(str(spack_dir))
+
+        assert exc_info.value.args[0] == errno.EACCES
+
+        # Ensure an OS Error is raised on a created user directory
+        spack_dir.ensure(dir=True)
+        user_dir = spack_dir.join(getpass.getuser())
+        with pytest.raises(OSError) as exc_info:
+            spack.stage._create_stage_root(str(user_dir))
+
+        assert exc_info.value.args[0] == errno.EACCES
+
+    @pytest.mark.nomockstage
+    def test_create_stage_root_bad_uid(self, tmpdir, monkeypatch):
+        """
+        Test the code path that uses an existing user path -- whether `$user`
+        in `$tempdir` or not -- and triggers the generation of the UID
+        mismatch warning.
+
+        This situation can happen with some `config:build_stage` settings
+        for teams using a common service account for installing software.
+        """
+        orig_stat = os.stat
+
+        class MinStat:
+            st_mode = -1
+            st_uid = -1
+
+        def _stat(path):
+            p_stat = orig_stat(path)
+
+            fake_stat = MinStat()
+            fake_stat.st_mode = p_stat.st_mode
+            return fake_stat
+
+        user_dir = tmpdir.join(getpass.getuser())
+        user_dir.ensure(dir=True)
+        user_path = str(user_dir)
+
+        # TODO: Once wrap os.stat or able to ensure using a pytest with
```

Comment:
I'm having trouble parsing this comment, should "or" -> "are"?

Reply:
No, maybe "or are".  Point is either wrap `os.stat`( as you'd suggested) or somehow ensure using a newer pytest.

Will re-word.